packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -109,14 +112,60 @@ export class MemoryStore implements Store {
     }
   }
 
-  public getBudget(site: string): Promise<SiteBudget | undefined> {
-    return Promise.resolve(this._budgets[site]);
+  public getBudget(site: string): Promise<SiteBudget> {
```

Comment:
curious, what's the reason for returning a `Promise`?

Reply:
It should probably be `async`, but every store function like this should return promises.